log/testutil.go

```diff
@@ -34,15 +34,42 @@ func interestingGoroutines() (gs []string) {
 			strings.Contains(stack, "stackimpact-go/internal") ||
 			// stackimpact uses persistent http client conns
 			strings.Contains(stack, "created by net/http.(*Transport).dialConn") ||
-			strings.Contains(stack, "log.MainTest") {
+			strings.Contains(stack, "log.MainTest") ||
+			matchesUserInterestingGoroutine(stack) {
 			continue
 		}
+
 		gs = append(gs, stack)
 	}
 	sort.Strings(gs)
 	return
 }
 
+var userInterestingGoroutines []string
+
+// AddUserInterestingGoroutine can be called when the environment of some
+// specific tests leaks goroutines unknown to interestingGoroutines()
```

Comment:
Add: `This function is not safe for concurrent execution. The caller should add of the desired exceptions before launching any goroutines.`